session.go

```diff
@@ -0,0 +1,160 @@
+type Session struct {
+	SessionID  string
+	Scopes     []string
+	OAuthState string
+	OIDCNonce  string
+	User       *User
+}
+
+// SessionStore is a map of #Session structs
+type SessionStore struct {
+	Store map[string]*Session
+}
+
+// Config is the OIDCMock server configuration information (here as placeholder)
+type Config struct {
+	ClientID     string
+	ClientSecret string
+	Keypair      *Keypair
+	AccessTTL    int64
+	RefreshTTL   int64
+	Issuer       string
+}
+
+// NewSession returns a pointer to a new Session struct
+func (ss *SessionStore) NewSession(scope string, oAuthState string, oidcNonce string, user *User) (*Session, error) {
+	sessionID, err := nonce(24)
+	if err != nil {
+		return nil, err
+	}
+
+	session := &Session{
+		SessionID:  sessionID,
+		Scopes:     strings.Split(scope, " "),
+		OAuthState: oAuthState,
+		OIDCNonce:  oidcNonce,
+		User:       user,
+	}
+
+	ss.Store[sessionID] = session
+
+	return session, nil
+}
+
+// AccessToken returns the JWT token with the appropriate claims for an access token
+func (s *Session) AccessToken(config *Config, timeFunc TimeNowFunc) (string, error) {
+
+	claims := standardClaims(s, config, config.AccessTTL, timeFunc)
+
+	token, err := config.Keypair.SignJWT(claims)
+	if err != nil {
+		return "", err
+	}
+
+	return token, nil
+}
+
+// standardClaims returns a populated jwt.StandardCLaims struct
+func standardClaims(s *Session, config *Config, ttl int64, timeFunc TimeNowFunc) jwt.StandardClaims {
+	if timeFunc == nil {
+		timeFunc = TimeFunc
+	}
+
+	return jwt.StandardClaims{
+		Audience:  "Audience",
+		ExpiresAt: timeFunc().UnixNano() + ttl,
+		Id:        s.SessionID,
+		IssuedAt:  timeFunc().UnixNano(),
+		Issuer:    config.Issuer,
+		NotBefore: timeFunc().UnixNano(),
+		Subject:   s.User.Email,
+	}
+}
+
+// RefreshToken returns the JWT token with the appropriate claims for an access token
+func (s *Session) RefreshToken(config *Config, timeFunc TimeNowFunc) (string, error) {
+	claims := standardClaims(s, config, config.RefreshTTL, timeFunc)
+
+	token, err := config.Keypair.SignJWT(claims)
+	if err != nil {
+		return "", err
+	}
+
+	return token, nil
+}
+
+// IDTokenClaims are the claims to be provided to jwt.Sign to create the ID token
+type IDTokenClaims struct {
+	PreferredUsername string   `json:"preferred_username,omitempty"`
+	Email             string   `json:"email,omitempty"`
+	Phone             string   `json:"phone_number,omitempty"`
+	Address           string   `json:"address,omitempty"`
+	Groups            []string `json:"groups,omitempty"`
+	EmailVerified     *bool    `json:"email_verified,omitempty"`
+	jwt.StandardClaims
+}
+
+// IDToken returns the JWT token with the appropriate claims for an access token
+func (s *Session) IDToken(config *Config, timeFunc TimeNowFunc) (string, error) {
+	claims := standardClaims(s, config, config.AccessTTL, timeFunc)
+
+	var idClaims IDTokenClaims
+
+	idClaims.Audience = "Audience"
```

Comment:
We can set these all at once:
```
claims := IDTokenClaims{
    StandardClaims: standardClaims(s, config, config.AccessTTL, timeFunc)
}
```

Reply:
Pretty sure you use the name of the composite struct (`jwt.StandardClaims`) directly since it doesn't have a field name.  I think you drop the `jwt` package part... but don't quote me on that 😅